### DISCUSSION TITLE
Simple BirdnetPi view on my website

### DISCUSSION THREAD

**Discussion by roelvanderkraan** [Nachtzuster/BirdNET-Pi#185] [Show and tell]
I've made a small page that displays the latest detections from my BirdnetPi on [my personal website](https://www.roelvanderkraan.nl/vogels). It makes the detections much more accessible for my partner and neighbours. It uses the Birdweather API to get the data.

See https://www.roelvanderkraan.nl/vogels

![Scherm­afbeelding 2024-09-08 om 12 00 18](https://github.com/user-attachments/assets/a6aaa12d-da33-44a9-8325-3f89878697a4)

It was inspired by the [Luistervink project](https://www.luistervink.nl/luistervink/maximapark/) in Utrecht (which is also how I found out about BirdnetPi)

**Comment by Svardsten53** [comment]
It looks very nice, and I would love to have something similar on my site. I assume it requires quite a bit of coding

**Reply by roelvanderkraan** [reply]
It wasn't too bad, I could write down how i did it if you're interested. Basically it uses Javascript to retrieve and display the bird data. The only complicating factor was that I didn't want to have my stations birdweather ID's in the public source code so I'm using Cloudflare workers as an intermediate between my website and Birdweather.

**Reply by Svardsten53** [reply]
Thanks, I'd be happy to receive code and instructions, and we'll see if I can adapt it to my own site.

**Reply by Kerakis** [reply]
I'd also be interested in doing something like this.

**Comment by jmtmp** [comment]
Thanks, that's exactly what I wanted to do for myself, but I didn't know how.

**Reply by jmtmp** [reply]
How about programming some new page on the pi server that would return xml/json data about the last N detections, an hourly summary for the last 24 hours and a monthly summary for the last 12 months.
This could be quite simple and everyone would use it as they like

**Reply by roelvanderkraan** [reply]
That might be possible, but the downside is that then your BirdnetPi should be accessible from the internet, which could require quite some setup like a fixed IP and port forwarding.. Thats why I am using Birdweather, they have an api that already returns this data and I dont have to open up my BirdnetPi to the outside world.

**Reply by Hiemstra87** [reply]
I have my PI's accessible from outside by installing Cloudflared on it and connecting it to some domain i own, fairly easy and safe.
